_episodes/10-prog.md

```diff
@@ -180,6 +180,85 @@ and translates any special characters in the values
 into their escaped equivalents
 so that they are safe to use.
 
+We can also use `sqlite3`'s cursor to make changes to our database,
+such as inserting a new name.
+For instance, we can define a new function called `add_name` like so:
+
+~~~
+def add_name(database_file, new_person):
+    query = "INSERT INTO Person VALUES" + repr(person_details) + ";"
+
+    connection = sqlite3.connect(database_file)
+    cursor = connection.cursor()
+    cursor.execute(query)
+    cursor.close()
+    connection.close()
+
+
+def get_name(database_file, person_id):
+    query = "SELECT personal || ' ' || family FROM Person WHERE id=?;"
+
+    connection = sqlite3.connect(database_file)
+    cursor = connection.cursor()
+    cursor.execute(query, [person_id])
+    results = cursor.fetchall()
+    cursor.close()
+    connection.close()
+
+    return results[0][0]
+
+# Insert a new name
+add_name('survey.db', ('barrett', 'Mary', 'Barrett'))
+# Check it exists
+print("Full name for barrett:", get_name('survey.db', 'barrett'))
+~~~
+{: .python}
+~~~
+IndexError: list index out of range
+~~~
+{: .output}
+
+The `get_name` command has failed, even though we added Mary's
+entry into the table using `add_name`.
+This is because we must perform a `connection.commit()` before closing
+the connection, in order to save our changes to the database.
```

Comment:
This will not actually fail on some versions of Python. The lack of autocommit is a recent change

Reply:
Thanks for pointing this out - I didn't realise this!

A little research has shown that sqlite3 became PEP 249 compliant after version 2.5, so I updated the surrounding text to provide a little more information for those running earlier versions.